File: agent/api/src/main/java/io/mockk/agent/MockKAgentFactory.java
package io.mockk.agent;

public interface MockKAgentFactory {
    void init(MockKAgentLogFactory logFactory);

    MockKInstantiatior getInstantiator();

    MockKProxyMaker getProxyMaker();

    MockKStaticProxyMaker getStaticProxyMaker();

    MockKConstructorProxyMaker getConstructorProxyMaker();
}

File: agent/api/src/main/kotlin/io/mockk/proxy/MockKAgentFactory.kt
package io.mockk.proxy

interface MockKAgentFactory {

    val instantiator: MockKInstantiatior

    val proxyMaker: MockKProxyMaker

    val staticProxyMaker: MockKStaticProxyMaker

    val constructorProxyMaker: MockKConstructorProxyMaker

    val interceptionScope: ProxyInterceptionScope

    fun init(logFactory: MockKAgentLogFactory)
}
